src-tauri/src/utils/utils.rs

```diff
@@ -88,3 +90,35 @@ fn extract_port(config: &Value, key: &str) -> Result<u32> {
         .and_then(|v| v.parse().ok())
         .ok_or_else(|| AppError::Custom(format!("Failed to extract {} port", key)))
 }
+
+pub fn is_port_in_use(port: u32) -> bool {
+    match format!("127.0.0.1:{}", port).parse::<SocketAddr>() {
+        Ok(addr) => TcpStream::connect_timeout(&addr, Duration::from_secs(1)).is_ok(),
+        Err(_) => false,
+    }
```

Comment:
```suggestion
    let Ok(addr) = format!("127.0.0.1:{}", port).parse::<SocketAddr>() else {
        return false;
    };
    
    TcpStream::connect_timeout(&addr, Duration::from_secs(1)).is_ok()
```

also, this is known to be valid (so long as `port` is a u16)

[`SocketAddr::new`](https://doc.rust-lang.org/nightly/core/net/enum.SocketAddr.html#method.new) + [`IpAddr: From<Ipv4Addr>`](https://doc.rust-lang.org/nightly/core/net/struct.Ipv4Addr.html#impl-From%3CIpv4Addr%3E-for-IpAddr)


```suggestion
    let addr = SocketAddr::new(Ipv4Addr::LOCALHOST.into(), port);
    TcpStream::connect_timeout(&addr, Duration::from_secs(1)).is_ok()
```

or [`SocketAddr: From<(I, u16)> where I: Into<IpAddr>`](https://doc.rust-lang.org/nightly/core/net/enum.SocketAddr.html#impl-From%3C(I,+u16)%3E-for-SocketAddr)

```suggestion
    let addr = SocketAddr::from((Ipv4Addr::LOCALHOST, port));
    TcpStream::connect_timeout(&addr, Duration::from_secs(1)).is_ok()
```